Library/Formula/fstar.rb

```diff
@@ -0,0 +1,69 @@
+class Fstar < Formula
+  desc "F*, an ML-like language with a type system for program verification"
+  homepage "http://fstar.gforge.inria.fr/"
+  url "https://github.com/FStarLang/FStar.git", :tag => "v0.9.0",
+                                                :revision => "f24457213f4007302873f6d42eb2cf8094fa45d0"
+  head "https://github.com/FStarLang/FStar.git", :branch => "master"
+
+  depends_on "mono"
+  depends_on "fsharp"
+  depends_on "ocaml"
+  depends_on "homebrew/science/z3"
+
+  resource "ocamlfind" do
+    url "http://download.camlcity.org/download/findlib-1.5.5.tar.gz"
+    sha256 "aafaba4f7453c38347ff5269c6fd4f4c243ae2bceeeb5e10b9dab89329905946"
+  end
+
+  resource "batteries" do
+    url "https://github.com/ocaml-batteries-team/batteries-included/archive/v2.3.1.tar.gz"
+    sha256 "df778b90fcdb26288d9d92a86e51dd75d6bb7c6e41888c748c7508e8ea58b1d4"
+  end
+
+  def ocaml_install_package(*)
```

Comment:
Just in-line this. It's not huge enough or used often enough to need its own block.

```ruby
resource("ocamlfind").stage do
  system "./configure"
  system "make", "all"
  system "make", "install"
end

resource("batteries").stage do
  system "make", "all"
  system "make", "install"
end
```